### PR TITLE
PWA-416 implement search route

### PR DESCRIPTION
**Pull request by richardgorman**

### CODE REVIEW

File: components/Navigator/components/Content/index.jsx [hunk 1]
```diff
@@ -1,48 +1,41 @@
...
 
 /**
- * The navigator content component.
- * @param {Object} props The component props.
- * @returns {JSX}
+ * The NavigatorContent component.
  */
-const Content = (props) => {
-  let currentTitle = null;
+class NavigatorContent extends Component {
```
Comment (devbucket) [inline components/Navigator/components/Content/index.jsx]: Make it at least a `PureComponent`. You only have primitive props.

File: components/Navigator/components/SearchButton/index.jsx [hunk 2]
```diff
@@ -2,19 +2,17 @@ import React, { Component } from 'react';
 import PropTypes from 'prop-types';
 import MagnifierIcon from '@shopgate/pwa-ui-shared/icons/MagnifierIcon';
 import Ripple from '@shopgate/pwa-ui-shared/Ripple';
+import NavigatorContext from '../../context';
 import styles from './style';
 
 /**
  * The search button component.
  */
 class SearchButton extends Component {
```
Comment (devbucket) [inline components/Navigator/components/SearchButton/index.jsx]: You should also make this a `PureComponent`. You only receive primitive props.

File: components/Navigator/index.jsx [hunk 3]
```diff
@@ -39,6 +44,80 @@ class Navigator extends PureComponent {
...
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      routePattern: '',
+      searchField: false,
+      searchQuery: '',
+    };
+
+    UIEvents.addListener('UI_NAVIGATOR_SEARCH_FIELD', this.toggleSearchField);
```
Comment (devbucket) [inline components/Navigator/index.jsx]: In my opinion, this enum should be a constant. Can you please change that here?

File: components/Navigator/components/SearchButton/index.jsx (hunk 2 above)
Comment (richardgorman) [reply components/Navigator/components/SearchButton/index.jsx]: In this case, there is an explicit check inside stopping the component from updating.

File: components/Navigator/components/Content/index.jsx (hunk 1 above)
Comment (richardgorman) [reply components/Navigator/components/Content/index.jsx]: Done

**Pull request merged**